### ISSUE TITLE
Error running pre-deployment DB migrate during switch from vanilla mastodon

### ISSUE THREAD

**Issue by Umiamz** [glitch-soc/mastodon#2924]
### Steps to reproduce the problem

1. Follow instructions to install glitch-soc
2. Reach point where first DB migrate is run
3. Migration quits with an error
...

### Expected behaviour

Migration completes successfully 

### Actual behaviour

Following error occurs

### Detailed description

I’m attempting to switch from vanilla Mastodon v4.4.0-alpha.1 at commit 8770905 (29/12/24). I’ve followed the instructions in the glitch-soc documentation and everything proceeds smoothly until the first DB migrate. This quits with the following error:

MoveGlitchUserSettings: migrating ===========================
bin/rails aborted!
StandardError: An error has occurred, all later migrations canceled: (StandardError)

PG::UndefinedColumn: ERROR:  column settings.thing_type does not exist
LINE 1: SELECT "settings".* FROM "settings" WHERE "settings"."thing_...

### Mastodon instance

mstdn.umiamz.me

### Mastodon version

v4.4.0-alpha.1 as of commit 8770905

### Environment

Please at least include those informations:
- Operating system: Debian 12
- Ruby version: 3.3.6p108
- Node.js version: 22.12.0

**Comment by ClearlyClaire** [comment]
Thanks for your report. This occurs because the migration is supposed to run much earlier, but doesn't due to the fact that you were not using glitch-soc. I'll think of a way to address that issue.

**Comment by Umiamz** [comment]
Thank you for the quick response and fix! I shall try the DB migration again this evening.

**Comment by Umiamz** [comment]
This time when I ran the DB migration there was no output. No error messages but no output after the "Loaded .env.production". I've checked the database and there is still no settings.thing_type column:

mastodon_production=# select "settings".* from "settings" where "settings"."thing_type" = "test";
ERROR:  column settings.thing_type does not exist

I'm wondering whether partially running the DB migration before you applied the fix has confused things. Is there a way to check whether the DB migration has made the necessary changes?
